### PR TITLE
PropertyUserStore should extract packed config file #1520

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: jetty-security/src/main/java/org/eclipse/jetty/security/PropertyUserStore.java [hunk 1]
```diff
@@ -111,18 +117,63 @@ public Path getConfigPath()
...
+                        while ( ( bytesRead = zin.read( buffer ) ) != -1 )
+                        {
+                            textOutputStream.write( buffer, 0, bytesRead );
+                        }
+                    }
+                    return extractedPath;
+                }
+            }
+        }
+        throw new RuntimeException( "cannot find file from url " + configFile );
```
Comment (janbartel) [inline jetty-security/src/main/java/org/eclipse/jetty/security/PropertyUserStore.java]: Or alternatively unpack the whole jar with 
JarResource r = JarResource.newJarResource(url);
r.copyTo(tmpDir)

If you do it with the ZipInputStream instead, I think you need to close the stream, no?

**Comment by olamy** [comment]
well @janbartel I have an issue using  JarResource.copyTo with such url  jar:file:/Users/olamy/dev/sources/open-sources/jetty/jetty.project/jetty-security/target/tests/testPropertyUserStoreLoadFromJarFile/users.jar!/mountain_goat/pale_ale.txt 
The file is not extracted. Doesn't look to be supported? JarFileResource seems to be the culprit.

**Comment by janbartel** [comment]
@olamy I meant use JarResource to unpack the whole jar.

**Comment by olamy** [comment]
@janbartel thanks for the tip! all done now.

**Pull request merged**